### ISSUE TITLE
Settings not persisting between IDE sessions on multiple devices. 

### ISSUE THREAD

**Issue by Neodern** [ChrisRM/material-theme-jetbrains#1356]
### Summary

Settings not persisting between IDE sessions on multiple devices. 

...

### Expected behavior

When i've for exemple "Material Oceanic" set, thesed should be persisted and be the theme when i open Intellij on an other device and if i'm also connected to my jetbrains account. 
...

### Actual behavior

I've 2 Computers connected to my jetbrains account, i set Material Oceanic Theme on Computer1, I Close IDEA (or webstorm) and i go to my computer 2, I start my IDE. The IDE is blank, and I've to go in Material UI Settings, change something, like check "Contrat Mode", apply, then uncheck "Contrat Mode", then i'm back with my Material Oceanic Theme.
I close IDE, go back to computer 1, start IDE, Material Oceanic quickly disappear (because i'm connected i suppose) and i go back with a ugly blank theme. I've to switch an option again in MaterialUI theme to get back my theme. each time i start my IDE ....

### Environment

IntelliJ IDEA 2019.2.1 (Ultimate Edition)
Build #IU-192.6262.58, built on August 20, 2019
Licensed to XX
Subscription is active until December 20, 2019
Runtime version: 11.0.3+12-b304.39 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.14.6
GC: ParNew, ConcurrentMarkSweep
Memory: 1450M
Cores: 12

![Capture d’écran 2019-08-29 à 09 39 29](https://user-images.githubusercontent.com/5598626/64000160-6c137400-cb05-11e9-8709-9aa3230b484c.png)

**Comment by Hvalfisk** [comment]
Disabling Settings Sync does make this bug go away.

**Comment by Neodern** [comment]
So i've to stop to synchronize all my configurations & plugins between 4 devices just because one plugin has a bug ? 
That's the solution ? O.o

**Comment by mallowigi** [comment]
OK I managed to reproduce.

**Comment by mallowigi** [comment]
Ok so @Neodern, if I understand correctly, you're switching themes via the options panel, is that it ?

**Comment by mallowigi** [comment]
Ok I can confirm that my plugin's settings are not synchronized. That's the core of the issue here.

Besides, changing the theme via the settings page will not change the current "Look And Feel" (even though it looks like it) so the last LAF takes precedence, in this case Darcula. If you changed from the Quick Switch though, it should work.

**Comment by Neodern** [comment]
So if i pick theme "Material Oceanic" from this menu, and i save. It should be ok on my other device ? 

![Capture d’écran 2019-09-11 à 09 31 42](https://user-images.githubusercontent.com/5598626/64676985-fe593780-d476-11e9-8d2b-cc90c2243b37.png)


I don't understand why if i change ... "Contrast Mode" for example, the Material Oceanic Theme is ok and applied with contrast mode. But if i go on an other device synchronized i've the material oceanic theme applied at started but quickly fallback to an ugly default light theme. :/ 

nvm, I applied Material Oceanic from dropdown menu, i will give it a try later on my other device to check if it's solved.

**Comment by mallowigi** [comment]
As I said I am seeing that my settings are not synchronized with the sync plugins. I don’t know why this is the case as I hadn’t read anything about it. I’ll try to look for more information.

The light theme is actually because of what you are seeing if you go to Settings > Appearance > Look and Feel. There is an issue about changing the theme from the Material Theme settings not changing the look and feel, hence not synchronized between accounts. But if you change from the actions panel or from the quick switch it should be working.

--
Elior

**Comment by mallowigi** [comment]
Found this: https://intellij-support.jetbrains.com/hc/en-us/community/posts/360000594130-IDE-Settings-not-synced
https://blog.jetbrains.com/idea/2017/09/intellij-idea-2017-3-eap-better-synchronization-of-your-settings-across-devices-and-other-improvements/

Don't know if this is still relevant but anyway it looks like settings synchronization for Material Theme and other plugins is not possible.

So I'm closing this issue.

**Comment by petrleocompel** [comment]
I have 4 devices with WebStorm, IDEA and etc.. 
Ubuntu, Win and 2 MacOS. Everytime when IDE sync it will reset. Most time it is visible between going from Ubuntu to MacOS and backwards (not using Win that much). But it resets even between MacOS to MacOS with Webstorm.
I'm using `Material Darker` so it's really easy to see it and it's easy to reproduce for me.
Isn't there a posilibity that saved settings is not same on all systems?

**Comment by mallowigi** [comment]
As I said before, sync between material theme settings is not possible. Look in the aforementioned link. However because the material theme plugin changes the look and feel and that look and feels are synced, so at least the theme is synced, but partly because it is not the same settings.

In short, font expect settings synchronization. It’s not supported at all.

**Comment by ForsakenHarmony** [comment]
So say I have material theme configured on 2 systems as I want it, not synced but not important

You can't do anything about it reverting to drakula when settings sync is on?

**Comment by mallowigi** [comment]
Not really, I’ll explain:

Sync settings are currently synchronizing all settings that are not third party, meaning look and feels as well. Material theme settings are third party settings so they are not synced, up to here everything is ok.

However since JB released themes, the Material UI Themes have been converted to look and feels as well, so even though MT settings are not synced, the look and feels are.

Here’s the tricky part: changing a theme from the material theme settings or action changes the current theme in Mt settings AND appearance settings, but because only appearance settings are synced, something out of sync happens, and JB is looking for a theme that is not yet loaded, such as Material Oceanic, fails and revert back to Darcula. Or not. It can finish loading the plugin, then sync, and find the Oceanic LaF and everything is fine.

In short it’s not consistent, and until there is no sync settings between plugins is impossible to fix.

**Comment by mallowigi** [comment]
Disabling only look and feel would solve the issue, I guess.

**Comment by zhigang1992** [comment]
![image](https://user-images.githubusercontent.com/1057756/71345543-a61af100-25a0-11ea-9143-cb6199fc0dc7.png)

It doesn't seem to be enough, I have both "Look and feels" and "Editor Color Schemes" disabled, and still get reverted back to the original theme. :(

I've disabled "Appearance & Behavior" as well now, will report back if that helps.
